### ISSUE TITLE
Head tracking

### ISSUE THREAD

**Issue by mzed** [immersive-audio-live/ADM-OSC#7]
Data from head tracking is useful for binaural representation of immersive audio. We should make a suggestion for how that data is represented in OSC, although that isn't part of ADM. 

There's a long discussion of representations here: https://research.mach1.tech/posts/describing-3d-motion/ 

Bridgehead sends out quaternions: https://supperware.co.uk/headtracker and Yaw-Pitch-Roll

We should make some suggestion for standardization.

**Comment by JPagnier** [comment]
Hi everyone 

I am up for restarting the discussion on the Yaw, Pitch, Roll parameters as well as the XYZ position of the listener and integrating this into the ADM-OSC so that third-party head tracking solution developers can immediately offer a standardized OSC output.

What can you do together?

**Comment by mzed** [comment]
One roadblock is that stakeholders wanted a clear relationship between ADM and ADM-OSC.  (Makes sense to me.). So, a problem with these messages -- as well as `/touch` or `/release` -- is they don't have a definable translation into ADM.

Maybe ADM-OSC should be renamed "ImmersiveAudio-OSC" or some other generic name?

I propose another namespace, like `/ctrl/tracker/ypr fff` that we could develop alongside of the `/adm` messages.  That would be an area to put control messages that don't directly translate to ADM, but would be very useful to the current users of ADM-OSC.

**Comment by JPagnier** [comment]
I agree with your analysis. There is indeed, a rather basic layer, of agreeing on common OSC terms, and there is indeed the ability to adhere to the ADM system

**Comment by madees** [comment]
Having a "control" unified name space closer to MIDI could be a good idea also, there is already some initiative like https://github.com/fabb/SynOSCopy/wiki

I've heard also that some are working on 6DOF, 9DOF... maybe this could be an addition to ADM in the future ?
https://www.bbc.co.uk/rd/publications/tools-six-depth-of-field-immersive-audiovisual-content-capture-production
IRCAM also. Worth to sync with EBU for updates ?

**Comment by madees** [comment]
this is the latest proposal on this topic : 
**/lis/    Head Tracking**
We saw more and more headphones renderer with headtracking use cases.
The proposal is to unify 6 DOF parameters for listener position and orientation.

The proposal could be :
/adm/lis/xyz f,f,f
/adm/lis/pyr f,f,f
Axis convention like above, pitch following x axis, yaw following y axis and roll z axis .


It is a key feature to allow rendering a scene from allocentric (XYZ physic-based position of objects) to egocentric (AED relative to single listener position) renderers and vis versa.

This isn’t currently supported in ADM, but may be in the future, as we see more and more allocentric object-based audio in live, 3D games, AR and VR, and interactivity where the listener can freely move in a physic-based environment.

If this information isn’t communicated, default could be that listener position is (0,0,0) and orientation (0,0,0), i.e. in the center of cube/sphere transformation and orientated forward, along Y axis.

MZ Looking at Supperware, it sends

/quaternion X Y Z W
/yaw float
/pitch float
/roll float

I don’t see a place in the supperware documentation where it specifies angles, but maybe this is good:

•	Yaw (left -> right --- where rotating left is negative)
•	Pitch (down -> up --- where rotating down is negative)
•	Roll (top-pointing-left --- top-pointing-right | where rotating top of object left is negative)

This is a good discussion: https://research.mach1.tech/posts/describing-3d-motion/ 

We also need to be clear if x, y, z are the same axes as our cartesian position.  I think they should be, but people coming from animation might call y up-and-down and z depth.  For ADM-OSC, z is always height.

**Comment by JGaudin** [comment]
Very good proposal !
I think it all make sense.

**Comment by EmptyMusicBox** [comment]
Hi i'm okay with that :)

**Comment by madees** [comment]
wouldn't it be more coherent with

/adm/lis/xyz f,f,f
/adm/lis/ypr f,f,f

Axis convention like above, yaw following z axis, pitch following x axis, and roll y axis.
•	Yaw (right -> left --- where rotating right is negative) **(as azimuth is this way)**
•	Pitch (down -> up --- where rotating down is negative) **(coherent to elevation)**
•	Roll (top-pointing-left --- top-pointing-right | where rotating top of object left is negative) **(??? is there a rule of thumbs for euler transformation that would be better orientation of those three vectors ?)**

**Comment by mzed** [comment]
I notice that ADM has a parameter called **headLocked** --- for objects that don't follow head tracking. Maybe we also want that as a per-object configuration message?

**Comment by madees** [comment]
> I don't understand what `/adm/lis/xyz` would mean in this context.
> 
> When we have two ways of sending the same data, it becomes possible that a sender only sends one way and the receiver only understands the other. (This happened at the first plugfest with `/x`, `/y`, and `/z` vs `/xyz`.) Maybe just `/ypr` is better?

As an example, in d&b DS100 the soundscape is rendered depending of object vs speaker relative positions, there is no listener position or orientation in the equation, as it is a large multiple listeners environment. If we want to render the same OBA scene with single listener renderer like binaural headphones, those distance and azimuth parameters are object vs listener positions, so we'll need this listener (x,y,z) position  and (y,p,r) orientation to compute azimuth and distance of objects relative to this specific listener.
Also, in metaverse, VR and game engine, the listener can travel thru the virtual audience area.
And even on TV broadcast, interactive contents for sport events for example where you can choose your place in the stadium seems to be a feature for OBA.

**Comment by mzed** [comment]
Aha, my mistake.  I was thinking `/xyz` was somehow the same data as `/ypr`.  Location and orientation makes perfect sense.

**Comment by SylvainThevenard** [comment]
Dear all,
I agree with Mathieu and this :

/adm/lis/xyz f,f,f
/adm/lis/ypr f,f,f
Axis convention like above, yaw following z axis, pitch following x axis, and roll y axis.
• Yaw (right -> left --- where rotating right is negative)
• Pitch (down -> up --- where rotating down is negative)
• Roll (tilt left -> tilt right --- where rotating down left is negative)

**Comment by mzed** [comment]
One more twist on this.

Supperware (and maybe other trackers) is using Tait-Bryant angles. See https://en.wikipedia.org/wiki/Euler_angles

**Comment by SylvainThevenard** [comment]
I use Supperware headtracker and it works as described here. I just do not remember if rotating direction of yaw is like we propose.
